Userland/Libraries/LibGL/Tex/TextureUnit.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <LibGL/Tex/Texture2D.h>
+
+namespace GL {
+
+class TextureUnit {
+public:
+    TextureUnit() = default;
+
+    void bind_texture_to_target(GLenum texture_target, const RefPtr<Texture>& texture);
+    void unbind_texture(GLenum texture_target);
+    const RefPtr<Texture>& bound_texture() const { return m_currently_bound_texture; }
```

Comment:
```suggestion
    const RefPtr<Texture>& bound_texture() const { return m_currently_bound_texture; }
    const RefPtr<Texture2D>& bound_texture_2d() const { return m_texture_target_2d; }
```
Maybe also add a convenience method to directly get the bound 2d texture? This would relieve us of doing casting magic in the context.

Reply:
Good idea